### PR TITLE
lift subsystem

### PR DESCRIPTION
**Pull request by Mboesn**

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -23,40 +26,46 @@
   private static final String kCustomAuto = "My Auto";
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/Robot.java:36]: first of all we want the code to be clean from comments, please remove all comments that created automaticly when creating the robot project.

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -23,40 +26,46 @@
...
   private final SendableChooser<String> m_chooser = new SendableChooser<>();
+  public static Lift lift;
 
   /**
-   * This function is run when the robot is first started up and should be
-   * used for any initialization code.
+   * This function is run when the robot is first started up and should be used
+   * for any initialization code.
    */
   @Override
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/Robot.java]: also here

File: src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -23,40 +26,46 @@
...
   public void robotInit() {
     m_chooser.setDefaultOption("Default Auto", kDefaultAuto);
     m_chooser.addOption("My Auto", kCustomAuto);
     SmartDashboard.putData("Auto choices", m_chooser);
+    lift = new Lift(new VictorSP(RobotMap.LIFT_LEFT_MOTOR), new VictorSP(RobotMap.LIFT_RIGHT_MOTOR),
+        new Encoder(RobotMap.LIFT_ENCODER_CHANNELA, RobotMap.LIFT_ENCODER_CHANNELB));
+
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/Robot.java]: names: LIFT_ENCODER_CHANNEL_A, LIFT_ENCODER_CHANNEL_B

File: src/main/java/frc/robot/Robot.java [hunk 4]
```diff
@@ -23,40 +26,46 @@
...
+   * items like diagnostics that you want ran during disabled, autonomous,
+   * teleoperated and test.
    *
-   * <p>This runs after the mode specific periodic functions, but before
-   * LiveWindow and SmartDashboard integrated updating.
+   * <p>
+   * This runs after the mode specific periodic functions, but before LiveWindow
+   * and SmartDashboard integrated updating.
    */
   @Override
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/Robot.java]: commants...

File: src/main/java/frc/robot/Robot.java [hunk 5]
```diff
@@ -23,40 +26,46 @@
...
+   * auto name from the text box below the Gyro
    *
-   * <p>You can add additional auto modes by adding additional comparisons to
-   * the switch structure below with additional strings. If using the
-   * SendableChooser make sure to add them to the chooser code above as well.
+   * <p>
+   * You can add additional auto modes by adding additional comparisons to the
+   * switch structure below with additional strings. If using the SendableChooser
+   * make sure to add them to the chooser code above as well.
    */
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/Robot.java]: and commants...

File: src/main/java/frc/robot/RobotMap.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/RobotMap.java]: delete commants

File: src/main/java/frc/robot/RobotMap.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Add your docs here.
+ */
+public class RobotMap {
+
+    public static final int LIFT_LEFT_MOTOR = 0;
+    public static final int LIFT_RIGHT_MOTOR = 1;
+    public static final int LIFT_ENCODER_CHANNELA = 0;
+    public static final int LIFT_ENCODER_CHANNELB = 1;
+
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/RobotMap.java]: i want you to make it more clear which value (like LIFT_LEFT_MOTOR) is port of PWM (motor controllers) and which values are ports of DIO (sensors).

File: src/main/java/frc/robot/Subsystems/Lift.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/Subsystems/Lift.java]: delete commants

File: src/main/java/frc/robot/Subsystems/Lift.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.SpeedController;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+/**
+ * Add your docs here.
+ */
+public class Lift extends Subsystem {
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/Subsystems/Lift.java:10]: also here

File: src/main/java/frc/robot/Subsystems/Lift.java [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  SpeedController liftMotorRight, liftMotorLeft;
+  Encoder encoder;
+  SpeedControllerGroup liftMotorGroup = new SpeedControllerGroup(this.liftMotorLeft, this.liftMotorRight);
+
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/Subsystems/Lift.java]: names: liftRightMotor, liftLeftMotor, liftEncoder
and make the SpeedControllerGroup ( = new SpeedControllerGroup(this.liftMotorLeft, this.liftMotorRight); ) in the class constractor after the client write which liftRightMotor and liftLeftMotor he use in the main class -> Robot.java

File: src/main/java/frc/robot/Subsystems/Lift.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    this.encoder = encoder;
+  }
+
+  public void setMotorSpeed(double power) {
+    liftMotorGroup.set(power);
+  }
+
+  public double encoderTicks() {
+    return encoder.get();
+  }
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/Subsystems/Lift.java:48]: we want to know what is the hight **(in meters)** the lift system got not in ticks, so make a math calculation (if you use parameters we dont know put then in robot map (yeah fine put them there) and make TODO there that tell us we need to masure them or find what are they when we will have robot...
also change the function name to getHight()

File: src/main/java/frc/robot/Subsystems/Lift.java [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+  SpeedControllerGroup liftMotorGroup = new SpeedControllerGroup(this.liftMotorLeft, this.liftMotorRight);
+
+  public Lift(SpeedController liftMotorLeft, SpeedController liftMotorRight, Encoder encoder) {
+    this.liftMotorLeft = liftMotorLeft;
+    this.liftMotorRight = liftMotorRight;
+    this.encoder = encoder;
+  }
+
+  public void setMotorSpeed(double power) {
+    liftMotorGroup.set(power);
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/Subsystems/Lift.java]: name: double speed (not power)

File: src/main/java/frc/robot/Subsystems/Lift.java [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+
+  public double encoderTicks() {
+    return encoder.get();
+  }
+
+  @Override
+  public void initDefaultCommand() {
+    // Set the default command for a subsystem here.
+    // setDefaultCommand(new MySpecialCommand());
+  }
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/Subsystems/Lift.java:53]: delete commants

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
great code, pay attention to my commants i think you need to start rechange things from robot class.

File: src/main/java/frc/robot/Robot.java (hunk 3 above)
Comment (esh4) [reply src/main/java/frc/robot/Robot.java]: Not using encoders...

File: src/main/java/frc/robot/RobotConstants.java [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    public static class Sensors {
+        /**
+         * The value that the potentiometer will multiply the faction it returns (how
+         * many times the potentiometer has turned). this is done to give the user a
+         * useful unit. it is currenlty set to the circumference of the tube connected
+         * to the potentiometer beacuse we desire the value it returns to be in meters
+         * and not in turns of the tube.
+         */
+        public static final double LIFT_POTENTIOMETER_SCALE_FRACTOR = RobotDimensions.LIFT_TUBE_CIRCUMFERENCE;
+        /**
```
Comment (esh4) [inline src/main/java/frc/robot/RobotConstants.java:34]: Factor or franction

File: src/main/java/frc/robot/Subsystems/Lift.java [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+   * This function checks whether the lift has activated the botton micro switch.
+   */
+  public boolean getBottomSwitch() {
+    return bottomSwitch.get();
+  }
+
+  /** This function returns the curent state of the potentiometer. */
+  public AnalogPotentiometer getPotentiometer() {
+    return this.potentiometer;
+  }
```
Comment (esh4) [inline src/main/java/frc/robot/Subsystems/Lift.java:48]: Returns a double

**Review by esh4 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Subsystems/Lift.java [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+  /**
+   * This function checks whether the lift has activated the botton micro switch.
+   */
+  public boolean getBottomSwitch() {
+    return bottomSwitch.get();
+  }
+
+  /** This function returns the curent state of the potentiometer. */
+  public Double getPotentiometer() {
+    return this.potentiometer.get();
```
Comment (esh4) [inline src/main/java/frc/robot/Subsystems/Lift.java:47]: double

**Review by esh4 — CHANGES_REQUESTED** [review]

**Review by esh4 — APPROVED** [review]

**Pull request merged**